pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -0,0 +1,399 @@
+func (Plugin) OnInboundCluster(env model.Environment, node model.Proxy, service *model.Service,
+	servicePort *model.Port, cluster *xdsapi.Cluster) {
+}
+
+// OnOutboundRouteConfiguration implements the Plugin interface method.
+func (Plugin) OnOutboundRouteConfiguration(in *plugin.InputParams, route *xdsapi.RouteConfiguration) {
+}
+
+// OnInboundRouteConfiguration implements the Plugin interface method.
+func (Plugin) OnInboundRouteConfiguration(in *plugin.InputParams, route *xdsapi.RouteConfiguration) {
+}
+
+// OnOutboundCluster implements the Plugin interface method.
+func (Plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service *model.Service,
+	servicePort *model.Port, cluster *xdsapi.Cluster) {
+}
+
+func isRbacEnabled(store model.IstioConfigStore) (bool, error) {
+	var configProto *rbacproto.RbacConfig
+	config := store.RbacConfig(RbacConfigName, kube.IstioNamespace)
+	if config != nil {
+		configProto = config.Spec.(*rbacproto.RbacConfig)
+	}
+	if configProto == nil {
+		return false, fmt.Errorf("couldn't find RbacConfig %s in namespace %s",
+			RbacConfigName, kube.IstioNamespace)
+	}
+	// TODO(yangminzhu): Supports ON_WITH_INCLUSION and ON_WITH_EXCLUSION.
+	if configProto.Mode != rbacproto.RbacConfig_ON {
+		log.Debugf("rbac plugin disabled by rbacConfig: %v", *configProto)
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// buildHTTPFilter builds the RBAC http filter that enforces the access control to the specified
+// service which is co-located with the sidecar proxy.
+func buildHTTPFilter(hostName model.Hostname, store model.IstioConfigStore) (*http_conn.HttpFilter, error) {
+	service := string(hostName)
+	split := strings.Split(service, ".")
+	if len(split) < 2 {
+		return nil, fmt.Errorf("failed to extract namespace from service: %s", service)
+	}
+	namespace := split[1]
+
+	roles := store.ServiceRoles(namespace)
+	if roles == nil {
+		return nil, fmt.Errorf("failed to get ServiceRoles in namespace %s", namespace)
+	}
+
+	bindings := store.ServiceRoleBindings(namespace)
+	if bindings == nil {
+		return nil, fmt.Errorf("failed to get ServiceRoleBinding in namespace %s", namespace)
+	}
+
+	log.Debugf("%s: converting RBAC rules to proxy config", RbacFilterName)
+	config, err := convertRbacRulesToFilterConfig(service, roles, bindings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert RBAC rules to filter config: %v", err)
+	}
+
+	return &http_conn.HttpFilter{
+		Name:   RbacFilterName,
+		Config: util.MessageToStruct(config),
+	}, nil
+}
+
+// convertRbacRulesToFilterConfig converts the current RBAC rules (ServiceRole and ServiceRoleBindings)
+// in service mesh to the corresponding proxy config for the specified service. The generated proxy config
+// will be consumed by envoy RBAC filter to enforce access control on the specified service.
+func convertRbacRulesToFilterConfig(service string, roles []model.Config, bindings []model.Config) (*rbacconfig.RBAC, error) {
```

Comment:
[optional] Move this function to model/, and simplifies it to smt like `getRbacFilterConfig(service string)` (i.e roles, bindings etc can be fetch internally).

Reply:
The output is the RBAC filter config that is tightly related to this RBAC plugin, not like the ServiceRole or ServieRoleBindings, so I prefer to put it here.